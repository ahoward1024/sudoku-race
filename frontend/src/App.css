.container {
  position: absolute;
  width: 80vw;
  padding-top: 100%;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.square-div {
  position: absolute;
  width: 100%;
  padding-top: 100%;
  height: auto;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
