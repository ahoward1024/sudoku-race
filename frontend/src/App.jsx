import React, { Component } from 'react';
import './App.css';
import Board from './Board';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      outer: {
        left: 10,
        top: 10,
        width: 0,
        height: 0
      },
      inner: {
        left: 20,
        top: 20,
        width: 0,
        height: 0,
      }
    };
    this.layout = this.layout.bind(this);
  }

  componentDidMount() {
    this.layout();
    window.addEventListener("resize", this.layout);
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this.layout);
  }

  layout() {
    var pad = 100;

    var head = document.getElementById("heading");
    var styles = window.getComputedStyle(head);
    var margin = parseFloat(styles["marginTop"]) + parseFloat(styles["marginBottom"]);
    var outer_w = window.innerWidth;
    var outer_h = window.innerHeight - (margin * 2);

    const base = 500;
    var scale = Math.min((outer_w - pad) / base, (outer_h - pad) / base);
    var size = base * scale;
    const max = 800;
    if(size > max) size = max;
    var inner_w = size;
    var inner_h = size;

    this.setState({
      outer: {
        left: 0, 
        top: margin * 2, 
        width: outer_w, 
        height: outer_h
      },
      inner: {
        left: ((outer_w / 2) - (inner_w / 2)), 
        top: ((outer_h / 2) - (inner_h / 2)),
        width: inner_w, 
        height: inner_h}
      }
    );
  }

  render() {
    return (
      <div>
        <h1 id="heading" align="center">Sudoku Race</h1>
        <div className="outer" style={this.state.outer}>
          <div id="square" className="inner" style={this.state.inner}>
            <Board width={this.state.inner.width} height={this.state.inner.height}/>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
