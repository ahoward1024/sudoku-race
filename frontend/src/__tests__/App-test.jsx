import React from 'react';
import {shallow} from 'enzyme';
import App from '../App';
import renderer from 'react-test-renderer';

test('Test the full App function', () => {
  const component = renderer.create(
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
