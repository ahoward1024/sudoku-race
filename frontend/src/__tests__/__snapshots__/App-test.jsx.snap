// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the full App function 1`] = `null`;
